test/triqs/utility/itertools.cpp

```diff
@@ -0,0 +1,128 @@
+/*******************************************************************************
+ *
+ * TRIQS: a Toolbox for Research in Interacting Quantum Systems
+ *
+ * Copyright (C) 2017 by O. Parcollet, N. Wentzell
+ *
+ * TRIQS is free software: you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option) any later
+ * version.
+ *
+ * TRIQS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * TRIQS. If not, see <http://www.gnu.org/licenses/>.
+ *
+ ******************************************************************************/
+#include <triqs/utility/itertools.hpp>
+#include <triqs/test_tools/arrays.hpp>
+#include <vector>
+
+using namespace triqs::utility;
+
+// An uncopyable int
+struct _int {
+  int i;
+  _int(int u) : i(u){};
+  _int(_int const&) = delete;
+  _int(_int&&)      = default;
+
+  operator int() const { return i; }
+  friend std::ostream& operator<<(std::ostream& out, _int const& x) { return out << x.i; }
+};
+
+TEST(Itertools, Zip) {
+
+  std::array<_int, 6> V1{6, 5, 4, 3, 2, 1};
+  std::vector<int> V2{1, 2, 3, 4, 5, 6};
+
+  // Zip both vectors and sum
+  for (auto[x, y] : zip(V1, V2)) {
+    TRIQS_ASSERT(7 - y == x);
```

Comment:
That makes a lot of sense! :) 

Thanks for the suggestion, I will update that.